neuros/main.py

```diff
@@ -47,125 +46,71 @@ def board_stream(board_id: int = BoardIds.SYNTHETIC_BOARD,
-    Windows can overlap based on configuration, useful for frequency analysis.
-
-    Args:
-        board: Connected and streaming BoardShim instance
-        config: Window configuration for size and overlap
-
-    Yields:
-        Windows of data as numpy arrays with shape (channels, samples)
-        where channels includes all board channels (both EEG and non-EEG)
-        and samples is determined by the window_ms configuration
-
-    Note:
-        The first few windows may be delayed as the board begins
-        collecting data. The generator will continue yielding
-        indefinitely until iteration is stopped.
-    """
-    # Convert ms-based config to samples
     sample_rate = board.get_sampling_rate(board_id=board.board_id)
     window_samples, overlap_samples = config.to_samples(sample_rate)
+    new_samples = window_samples - overlap_samples
 
-    # Buffer for overlap samples
-    buffer = np.array([])
+    # Get the EEG channel indices
+    eeg_channels = board.get_eeg_channels(board_id=board.board_id)
 
-    # Placeholder for new data
-    new_data = np.array([])
+    # Create buffer just for EEG channels
+    ring_buffer = np.zeros((len(eeg_channels), window_samples), dtype=np.float32)
+    filled_samples = 0
 
     try:
         while True:
-            # Calculate how many new samples we need
-            if buffer.size == 0:
-                samples_needed = window_samples
-            else:
-                samples_needed = window_samples - overlap_samples
+            try:
+                new_data = board.get_current_board_data(new_samples)
+                if new_data.size == 0 or new_data.shape[1] != new_samples:
+                    continue  # Skip if we don't have enough samples yet, e.g. at start
 
-            # Initialize new_data as empty but with correct shape
-            new_data = np.array([])
+                # Extract just the EEG channels from the new data
+                eeg_data = new_data[eeg_channels]
 
-            try:
-                # Get new data
-                new_data = board.get_current_board_data(samples_needed)
-                if new_data.size == 0:
-                    # No new data available yet
-                    continue
-
-                # Add to buffer
-                buffer = (np.concatenate([buffer, new_data], axis=1) if buffer.size > 0 else new_data)
-
-                # Once we have enough samples, yield a window
-                if buffer.shape[1] >= window_samples:
-                    # Extract the window
-                    window = buffer[:, :window_samples]
-                    yield window
-
-                    # Update buffer for next iteration
-                    if overlap_samples > 0:
-                        buffer = buffer[:, -overlap_samples:]
-                    else:
-                        buffer = np.array([])
+                ring_buffer[:, :-new_samples] = ring_buffer[:, new_samples:]
+                ring_buffer[:, -new_samples:] = eeg_data
```

Comment:
I don't think this code works as you'd expect.

Some things to check:
  - is new_data always equal to the new_samples in size?
  - what happens to the data in-between? I mean, the first line copies the tail end of the ring_buffer to the beginning. But then the next like copies eeg_data into the tail end vs where the new_samples ended. Perhaps you meant to leave out the minus sign?
  - would a test reveal an error in this copying?